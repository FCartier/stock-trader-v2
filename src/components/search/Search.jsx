import React, { Component } from "react";
import Select from "react-virtualized-select";
import "react-virtualized-select/styles.css";
import { handleSearch, getSymbols } from "./searchActions";
import { connect } from "react-redux";
import formatSearchData from "./utils/formatSearchData";
import * as searchIcon from "./images/search.png";

class Search extends Component {
  state = {
    selectedOption: null
  };

  componentDidMount() {
    this.props.getSymbols();
  }

  handleChange = selectedOption => {
    this.setState({ selectedOption });
    if(selectedOption)
      this.props.handleSearch(selectedOption.value);
  };

  render() {
    const { selectedOption } = this.state;
    return (
      <div className="full-height">
        <div className="icon-container">
          <img src={searchIcon} alt="search icon" className="search-icon" />
        </div>
        <div className="searchbox-container">
          <Select
              value={selectedOption}
              onChange={this.handleChange}
              options={this.props.results}
              id="select-component"
            />
        </div>
         
      </div>
    );
  }
}

const mapStateToProps = state => {
  console.log(state.search.symbols)
  console.log(formatSearchData(state.search.symbols))
  return {
    results: formatSearchData(state.search.symbols)
  }
}

const mapDispatchToProps =  {
    handleSearch,
    getSymbols
}

export default connect(mapStateToProps, mapDispatchToProps)(Search);
