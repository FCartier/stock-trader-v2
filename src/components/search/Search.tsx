import React, { Component } from "react";
import Select from "react-virtualized-select";
import "react-virtualized-select/styles.css";
import { selectedSymbol, SelectedSymbol } from "./searchActions";
import { connect } from "react-redux";
import formatSearchData from "./utils/formatSearchData";
import "./style.css"
import { Route } from 'react-router-dom'

/* tslint:disable */
import * as searchIcon from "./images/search.png";
import { any } from "prop-types";
import { search } from "./searchReducer";


interface ISearchProps {
  results: any;
  selectedSymbol: (input: string) => void;
}

interface ISearchState {
  selectedOption: null | object
}

class Search extends Component<ISearchProps, ISearchState> {
  constructor(props: ISearchProps) {
    super(props)
    this.state = {
      selectedOption: null
    }
  }


  public handleSubmit = (history: any, selectedOption: any) => {
    this.setState({
      selectedOption: selectedOption.value
    })
    history.push(`${selectedOption.value}`)
  };

  public render() {
    const { selectedOption } = this.state;
    return (
      <div className="full-height">
        <div className="icon-container">
          <img src={searchIcon} alt="search icon" className="search-icon" />
        </div>
        <div className="searchbox-container">
        <Route render={({history}) => (
         <Select
           value={selectedOption}
           onChange={(selectedOption) => this.handleSubmit(history, selectedOption)}
           options={this.props.results}
           id="select-component"
          />
         )}/>    
        </div>
      </div>
    );
  }
}

interface stateToProps {
    search: { 
      symbols: [], 
      symbol: string
    }
  
}

const mapStateToProps = (state: stateToProps) => {
  return {
    results: formatSearchData(state.search.symbols),
    symbol: state.search.symbol
  }
}

const mapDispatchToProps = {
  selectedSymbol
}

interface IMapProps {
  selectedSymbol?: (input: string) => SelectedSymbol;
}

interface IMapState {
  results: Array<{ value: string, label: string }>
}

export default connect<IMapState, IMapProps, {}>(mapStateToProps, mapDispatchToProps)(Search);
