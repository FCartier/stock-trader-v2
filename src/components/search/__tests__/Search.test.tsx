import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import { Search } from "../Search";
import formattedSearchData from "../__data__/formattedSearchData";

configure({ adapter: new Adapter() });

describe("search", () => {
  test("should render properly", () => {
    const component = shallow(
      <Search results={formattedSearchData} symbol="" selectedSymbol={jest.fn()} />
    );
    expect(component).toMatchSnapshot();
  });
});
