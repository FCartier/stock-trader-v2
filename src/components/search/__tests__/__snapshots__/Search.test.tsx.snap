// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search should render properly 1`] = `
<div
  className="full-height"
>
  <div
    className="icon-container"
  >
    <img
      alt="search icon"
      className="search-icon"
      src="test-file-stub"
    />
  </div>
  <div
    className="searchbox-container"
  >
    <Route
      render={[Function]}
    />
  </div>
</div>
`;
